runtime/ratelimit.c

```diff
@@ -215,7 +215,7 @@ ratelimitMsg(ratelimit_t *ratelimit, smsg_t *pMsg, smsg_t **ppRepMsg)
 	/* Only the messages having severity level at or below the
 	 * treshold (the value is >=) are subject to ratelimiting. */
 	if(ratelimit->interval && (pMsg->iSeverity >= ratelimit->severity)) {
-		if(withinRatelimit(ratelimit, pMsg->ttGenTime) == 0) {
+        if(withinRatelimit(ratelimit, pMsg->ttGenTime, getAPPNAME(pMsg, 0)) == 0) {
```

Comment:
I am not sure if it is a good idea to just pull the APPNAME. This is generic code here. So if e.g. a message is received from a remote system which is under ratelimiting, we get that remote system's APPNAME. I think that would be pretty confusing. Maybe it would be a good idea to add a source indication?